### DISCUSSION TITLE
MapLibre Android 11.5.0 API incompatibility icon padding

### DISCUSSION THREAD

**Discussion by louwers** [maplibre/maplibre-native#2873] [General]
Looks like MapLibre Android 11.5.0 introduces a backwards-incompatible when support for an array for `icon-padding` was merged https://github.com/maplibre/maplibre-native/pull/2845.  Apologies for this, it was an oversight on my part, we aim to stick to strict semantic versioning.

The incompatible APIs are:

 - `SymbolLayer.getIconPadding()` now returns a `Float[]` instead of a `Float`.
- `PropertyFactory.iconPadding(padding)` now requires a `Float[]` instead of a `Float` as parameter.

I can either revert the change in a patch release (technically making another breaking release), or let users migrate to the new API using the migration guide below.

## Migration Guide

- Replace `SymbolLayer.getIconPadding()` with `SymbolLayer.getIconPadding()[0]`.
- Replace `PropertyFactory.iconPadding(value)` with `PropertyFactory.iconPadding(new Float[]{value})`.

**Comment by random3940** [comment]
Opened #2880 to add PropertyFactory.iconPadding(Float) back.
